integration/tests/identity_provider/identity_provider_test.go

```diff
@@ -0,0 +1,379 @@
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/pkg/errors"
+
+	// Register the OIDC provider
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/sethvargo/go-password/password"
+
+	. "github.com/weaveworks/eksctl/integration/matchers"
+	. "github.com/weaveworks/eksctl/integration/runner"
+
+	"github.com/weaveworks/eksctl/integration/tests"
+	"github.com/weaveworks/eksctl/integration/utilities/kube"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+)
+
+const (
+	oidcGroupName = "oidc-reader"
+)
+
+var (
+	params                  *tests.Params
+	oidcConfig              *OIDCConfig
+	cleanupCognitoResources func() error
+)
+
+type OIDCConfig struct {
+	clientID     string
+	idToken      string
+	refreshToken string
+	idpIssuerURL string
+}
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	params = tests.NewParams("identity-provider")
+}
+
+func TestIdentityProvider(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = BeforeSuite(func() {
+	if !params.SkipCreate {
+		cmd := params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--verbose", "4",
+			"--name", params.ClusterName,
+			"--kubeconfig", params.KubeconfigPath,
+			"--nodes", "1",
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+
+	var err error
+	fmt.Fprintf(GinkgoWriter, "creating Cognito OIDC provider\n")
+	oidcConfig, err = setupCognitoProvider(params.ClusterName, params.Region)
+	Expect(err).NotTo(HaveOccurred())
+	fmt.Fprintf(GinkgoWriter, "created Cognito provider; client ID: %s\n", oidcConfig.clientID)
+})
+
+var _ = Describe("(Integration) [Identity Provider]", func() {
+
+	It("should associate, get and disassociate identity provider", func() {
+		By("associating a new identity provider")
+		identityProviderClusterConfig := makeIdentityProviderClusterConfig(oidcConfig, params.ClusterName, params.Region)
+
+		cmd := params.EksctlCmd.WithArgs(
+			"associate",
+			"identityprovider",
+			"--config-file", "-",
+			"--verbose", "4",
+			"--wait",
+		).
+			WithStdin(strings.NewReader(identityProviderClusterConfig)).
+			WithoutArg("--region", params.Region).
+			WithTimeout(40 * time.Minute)
+
+		Expect(cmd).To(RunSuccessfully())
+
+		By("getting the identity provider")
+		cmd = params.EksctlGetCmd.
+			WithArgs(
+				"identityprovider",
+				"--cluster", params.ClusterName,
+				"-o", "yaml",
+			)
+		Expect(cmd).To(RunSuccessfullyWithOutputStringLines(
+			ContainElement(ContainSubstring(fmt.Sprintf("ClientID: %s", oidcConfig.clientID)))),
+		)
```

Comment:
can we do more than this to check that its actually associated correctly? E.g. use the functionality, or query some other API to confirm it works.

Reply:
We do use the functionality below by constructing a Kubernetes client with the OIDC auth provider and ensuring it can access cluster resources.